Add capabilities table to READMEs

tore-espressif opened the pull request.

espzav commented on bsp/esp32_s3_usb_otg/README.md:
```diff
@@ -23,3 +23,18 @@ ESP32-S3-USB-OTG is a development board that focuses on USB-OTG function verific
 <!-- Autogenerated end: Dependencies -->
+<!-- Autogenerated start: Capabilities -->
+### Capabilities
+|   component   |version|
```
Rename header fields. For example: "Capability", "Available"

espzav reviewed (commented):
@tore-espressif Thank you for this PR! This is exactly right, what I meant.I left some comments about output design. 
I tried make review python code too, it looks good for me, but I'm still not good person for review python code. I am so sorry. 

I didn't test it.

espzav reviewed (commented):
I have one thought for discussion. Is it possible to combine capabilities and dependencies for get something like this:

|   Capability | Available | Type |
|---------------|-------|-------|
|DISPLAY        |   :heavy_check_mark:   | ILI9341 |
|TOUCH          |   :heavy_check_mark:   | TT21100 |
|BUTTONS        |   :heavy_check_mark:   |  - |
|AUDIO          |   :heavy_check_mark:   | - |
|AUDIO_SPEAKER  |   :heavy_check_mark:   | ES8311 |
|AUDIO_MIC      |   :heavy_check_mark:   | ES7210 |
|SDCARD         |   :x:   | - |
|IMU            |   :heavy_check_mark:   | icm42670  |

I know, that find this data is really hard. For example the LCD is not every used from dependencies, sometimes is from IDF. Audio is only i C file. I think, that there can be solution. Question is, how much time it takes in compare with benefit.

espzav commented on bsp/esp32_s3_usb_otg/README.md:
```diff
@@ -23,3 +23,18 @@ ESP32-S3-USB-OTG is a development board that focuses on USB-OTG function verific
+### Capabilities
+|   component   |version|
+|---------------|-------|
+|DISPLAY        |   1   |
```
I don't know, if it will work in component manager readme.

tore-espressif commented:
@espzav I fixed all the issues so the PR is ready now.

Your proposal with only 1 table that would have both dependencies and capabilities listed is great!
I just do not see a clear way of implementing it, for the reasons you mentioned 

>  For example the LCD is not every used from dependencies, sometimes is from IDF. Audio is only i C file.

I'll keep it hanging here for a moment and we can revisit it later

tore-espressif commented:
@espzav In commit 610ac58c92179834bf041ff3ca9d333a661a87b6 current dependecies table in README.md is extended to also include capabilities. The changes in README are in commit 3fe24b3a2f67fb089cbb41af7ef6ea225a6adaa8

PTAL whether the format of the table looks good to you

ATM, only BSPs with `BSP_CAPS_*` macros are updated

## TODO
- [ ] Fix related `noglib` CI job

espzav commented:
@tore-espressif Thank you for these changes, it is really great! I like this change! 
I didn't see code yet. Some nitpicks:
1. Missing some title above the table
2. It's a pitty, that we cannot read LCD types, when are used from IDF. Same for audio codecs.

tore-espressif commented:
@espzav Changes since your review:
1. Added table header
2. Fixed related noglib CI job
3. Improved error handling
> It's a pitty, that we cannot read LCD types, when are used from IDF. Same for audio codecs.

Yes, this information is not easily parsable :(

espzav reviewed (approved):
@tore-espressif Thank you for this work! LGTM
